pybop/problems/base_problem.py

```diff
@@ -27,6 +27,8 @@ class BaseProblem:
         Additional variables to observe and store in the solution (default: []).
     initial_state : dict, optional
         A valid initial state (default: None).
+    infeasible_solutions: bool, optional
+        If True, infeasible parameter values will be returned as normal evaluations (default: True).
```

Comment:
this docstring is not clear. returned from what? what is the difference between a normal evaluation and a not-normal evaluation in terms of what is returned?